eclass/freedict.eclass

```diff
@@ -1,37 +1,32 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 # @ECLASS: freedict.eclass
 # @MAINTAINER:
 # maintainer-needed@gentoo.org
 # @AUTHOR:
 # Original author: Seemant Kulleen
-# @SUPPORTED_EAPIS: 6
+# @SUPPORTED_EAPIS: 7
 # @BLURB: Ease the installation of freedict translation dictionaries
 # @DESCRIPTION:
 # This eclass exists to ease the installation of freedict translation
-# dictionaries.  The only variables which need to be defined in the actual
-# ebuilds are FORLANG and TOLANG for the source and target languages,
-# respectively.
+# dictionaries.
 
-# @ECLASS-VARIABLE: FORLANG
+# @ECLASS-VARIABLE: FREEDICT_P
 # @DESCRIPTION:
-# Please see above for a description.
-
-# @ECLASS-VARIABLE: TOLANG
-# @DESCRIPTION:
-# Please see above for a description.
+# Strips PN of 'freedict' prefix, to be used in SRC_URI and doins
+FREEDICT_P=${PN/freedict-/}
 
 case ${EAPI:-0} in
-	6) ;;
+	7)
+		[[ ${FORLANG} ]] && die "FORLANG is banned, set DESCRIPTION instead"
+		[[ ${TOLANG} ]] && die "TOLANG is banned, set DESCRIPTION instead"
+		;;
 	*) die "${ECLASS}.eclass is banned in EAPI=${EAPI}" ;;
 esac
 
-MY_P=${PN/freedict-/}
-
-DESCRIPTION="Freedict for language translation from ${FORLANG} to ${TOLANG}"
```

Comment:
As far as I can see, `DESCRIPTION` is no longer assigned in EAPI 6, and the existing ebuilds don't assign it either.